### PR TITLE
Add clone method for all ol/layers classes

### PR DESCRIPTION
**Pull request by oterral**
# How to

I propose to add clone functions to layers. It saves some code when layers have only one different property.
This PR just add a read only options property to the layers and a clone function on all layers.

# Others

- [x] It's not a hack or at least an unauthorized hack :).
- [x] The images added are optimized.
- [x] Everything in ticket description has been fixed.
- [x] The author of the MR has made its own review before assigning the reviewer.
- [x] The title means something for a human being.
- [x] The title contains [WIP] if it's necessary.
- [x] Labels applied. if it's a release? a hotfix?
- [x] The new class' members & methods are well documented
- [x] Tests added.

### CODE REVIEW

**Review by friedjoff — APPROVED** [review]
LGTM

🥳 for tests and consistency!

File: src/ol/layers/MapboxLayer.js [hunk 1]
```diff
@@ -326,7 +326,14 @@ export default class MapboxLayer extends Layer {
     // feature to be consistent with other layers.
     const features = this.mbMap
       .queryRenderedFeatures(pixel, options)
-      .map((feature) => this.format.readFeature(feature));
+      .map((feature) => {
+        const olFeature = this.format.readFeature(feature);
+        // We save the original mapbox feature to avoid losing informations
+        // potentially needed for other functionnality like highlighting
+        // (id, layer id, source, sourceLayer ...)
+        olFeature.set('mapboxFeature', feature);
```
Comment (friedjoff) [inline src/ol/layers/MapboxLayer.js:334]: Should we document and test this feature (no pun intended 😄)? Or do you see this only as a private property?

**Review by friedjoff — APPROVED** [review]
Still LGTM!

**Review by friedjoff — APPROVED** [review]
Documentation also looks good :)

File: src/ol/layers/MapboxLayer.js (hunk 1 above)
Comment (oterral) [reply src/ol/layers/MapboxLayer.js]: Fixed as discussed

**Pull request merged**